nuitka/build/CCompilerVersion.scons

```diff
@@ -175,6 +175,10 @@ if "CXX" in os.environ:
 # Patch the compiler detection.
 Environment.Detect = getEnhancedToolDetect()
 
+# Because after call `createEnvironment` os.environ will change.
```

Comment:
Duplicated code in Nuitka, very bad, still gradually fighting it.

Reply:
I see.. If I had a bit more free time right now (currently busy building a small utility and working on a couple of parsers, also delving into MSVC a bit), I would probably consolidate some functionality into SCons files to avoid duplication. I recommend adding ruff as an additional linter to black (a really useful and fast linter).